task5/script.js

```diff
@@ -0,0 +1,41 @@
+const df = document.createDocumentFragment();
+
+const ul = document.getElementById("ul");
+
+const listItem = [
+  {
+    to: "bookmark.html",
+    img: "1.png",
+    alt: "画像1",
+    text: "ブックマーク"
+  },
+  {
+    to: "message.html",
+    img: "2.png",
+    alt: "画像2",
+    text: "メッセージ"
+  }
+];
+
+function arrayPromise() {
+  return new Promise((resolve, reject) => {
+    resolve();
+  });
+}
+
+arrayPromise().then(() => {
+  for (let i = 0; i < listItem.length; i++) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = listItem[i].to;
+    a.textContent = listItem[i].text;
+
+    img.src = listItem[i].img;
+    img.alt = listItem[i].alt;
+
+    df.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
+  }
+  ul.appendChild(df);
+});
```

Comment:
I'll comment on the above review as adapted.
The `listItem` passed as the resolved value can be used in then.
```suggestion
arrayPromise(value).then(() => {
//Now try `console.log(value);` to see if it works.
  for (let i = 0; i < listItem.length; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");

    a.href = listItem[i].to;
    a.textContent = listItem[i].text;

    img.src = listItem[i].img;
    img.alt = listItem[i].alt;

    df.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
  }
  ul.appendChild(df);
});
```
Try to improve the loop here using the value passed to you.
We used `value` for the sake of argument, but you can name this part arbitrarily. I wrote it that way because I often use `value` as the value for now.

Reply:
I tried to output to the console.
```
[Object,Object]

0: Object
to: "bookmark.html"
img: "1.png"
alt: "画像1"
text: "ブックマーク"

1: Object
to: "message.html"
img: "2.png"
alt: "画像2"
text: "メッセージ"
```
After storing it in resolve, I understand that this is the resolved value.

Reply:
I'm sorry, I gave you the wrong place.😫
I'm really sorry for the confusion.

Not like this.
```js
arrayPromise(value).then(() => {
```
This is it.
```js
arrayPromise().then((value) => {
```
I'll write down what I wish I had done.
```js
arrayPromise().then((value) => {
  for (let i = 0; i < value.length; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.href = value[i].to;
    a.textContent = value[i].text;
    img.src = value[i].img;
    img.alt = value[i].alt;
    df.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
  }
  ul.appendChild(df);
});
```
[codesandbox](https://codesandbox.io/s/weathered-leaf-jd5m5?file=/script.js)

The reason why this code is not optimal is because
```js
arrayPromise().then(() => {
  for (let i = 0; i < listItem.length; i++) {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");
    a.href = listItem[i].to;
    a.textContent = listItem[i].text;
    img.src = listItem[i].img;
    img.alt = listItem[i].alt;
    df.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
  }
  ul.appendChild(df);
});
```
This is because we are referring to this constant.
This means that we're not using the resolved value.
```js
const listItem = [
  {
    to: "bookmark.html",
    img: "1.png",
    alt: "画像1",
    text: "ブックマーク"
  },
  {
    to: "message.html",
    img: "2.png",
    alt: "画像2",
    text: "メッセージ"
  }
];
```

Reply:
Thank you for the review.
I corected it!

I investigated the meaning of giving arguments to "resolve" and "then" methods, but I can't understand it.
Discord will ask you a question in Japanese.